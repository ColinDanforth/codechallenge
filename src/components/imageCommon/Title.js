import React from 'react'
import PropTypes from 'prop-types'
import DescriptionObject from "./DescriptionObject"

const rootTitleStyle={
  display: 'flex',
  flexDirection: 'Column',
  justifyContent: 'center',
  alignItems: 'center',
  minWidth: '600px',
  width: 'auto',
  height: 'auto',
  backgroundColor: '#FFF',
  borderBottom: 'inset',
}

const Title = (props) => {
  if(props.name !== undefined && props.name !== null && props.name.length > 0) {
    return (
      <div style={rootTitleStyle}>
        <h1>{props.name}</h1>
        <DescriptionObject description={props.description}/>
      </div>
    )
  } else {
    return (
      <div style={rootTitleStyle}>
        <h1>untitled</h1>
        <DescriptionObject description={props.description}/>
      </div>
    )
  }
}

export default Title

Title.propTypes={
  name: PropTypes.string,
  description: PropTypes.string,

}
