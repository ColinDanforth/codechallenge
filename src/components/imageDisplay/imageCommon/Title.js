import React from 'react'
import PropTypes from 'prop-types'

const rootTitleStyle={
  display: 'flex',
  flexDirection: 'Column',
  justifyContent: 'center',
  alignItems: 'center',
  minWidth: '600px',
  width: 'auto',
  height: 'auto',
  backgroundColor: '#FFF',
  border: 'inset',
}

const Title = (props) => {
  return(
    <div style={rootTitleStyle}>
      <h1>{props.name}</h1>
    </div>
  )
}

export default Title

Title.propTypes={
  name: PropTypes.string,
}
