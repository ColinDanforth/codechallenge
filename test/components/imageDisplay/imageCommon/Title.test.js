import React from 'react';
import Title from '../../../../src/components/imageDisplay/imageCommon/Title';
import renderer from 'react-test-renderer';

test('Title should display input test string, and description block with the same text', () => {
  const component = renderer.create(
    <Title name="Test String" description="Test String"/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

test('Title should display input test string, and a div in place of description with style of display none', () => {
  const component = renderer.create(
    <Title name="Test String" description=""/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

test('Title should display untitled, and a div in place of description with style of display none', () => {
  const component = renderer.create(
    <Title name="" description=""/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
