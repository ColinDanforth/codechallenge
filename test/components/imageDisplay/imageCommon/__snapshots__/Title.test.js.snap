// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should display input test string, and a div in place of description with style of display none 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderBottom": "inset",
      "display": "flex",
      "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
      "flexDirection": "Column",
      "height": "auto",
      "justifyContent": "center",
      "minWidth": "40vw",
      "width": "auto",
    }
  }
>
  <h1>
    Test String
  </h1>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;

exports[`Title should display input test string, and description block with the same text 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderBottom": "inset",
      "display": "flex",
      "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
      "flexDirection": "Column",
      "height": "auto",
      "justifyContent": "center",
      "minWidth": "40vw",
      "width": "auto",
    }
  }
>
  <h1>
    Test String
  </h1>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test String",
      }
    }
    style={
      Object {
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "height": "auto",
        "margin": "3%",
        "width": "90%",
      }
    }
  />
</div>
`;

exports[`Title should display untitled, and a div in place of description with style of display none 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderBottom": "inset",
      "display": "flex",
      "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
      "flexDirection": "Column",
      "height": "auto",
      "justifyContent": "center",
      "minWidth": "40vw",
      "width": "auto",
    }
  }
>
  <h1>
    untitled
  </h1>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;
